rslib/src/backend/collection.rs

```diff
@@ -150,6 +153,17 @@ impl CollectionService for Backend {
 }
 
 impl Backend {
+    fn lock_collection(&self, expected_open: bool) -> Result<MutexGuard<Option<Collection>>> {
```

Comment:
Just in case you're wondering, in the case of https://github.com/ankitects/anki/commit/d100f7a2c8681b2d4dd1bf03797ef8fb3157358b, a bool felt more appropriate, as I want the caller to consider whether syncing is appropriate or not, and it felt like that might be more easily overlooked with separate methods.

Also the #[cfg] change is an unnecessary micro-optimization - it cuts the block out at compile time, so no run-time check is needed.